### ISSUE TITLE
[BUG] RWX prior existing PV/PVC that were marked as XFS will now fail to mount, since we only supported EXT4 previously so we ignored users choice

### ISSUE THREAD

**Issue by joshimoo** [longhorn/longhorn#2991]
**Describe the bug**
Prior existing PV/PVC RWX that were marked as XFS will now fail to mount, since we only supported EXT4 at the time.
So we ignored the fs parameter from the PV, now that we include the FS param, the previously incorrectly created PV/PVC will now be tried to be mounted as XFS even though they contain an EXT4 filesystem.

This only impacts RWX volumes.

**To Reproduce**
1. On < v1.2 create a RWX volume manually via the UI
2. Create PV/PVC for that volume specify `XFS`
3. Spin up a workload once, so the volume gets formatted as `EXT4`
4. Update longhorn to v1.2 now the volume will fail to mount on the share-manager since it contains an `EXT4` filesystem but the desired filesystem was specified as `XFS`

**Expected behavior**
The volume should have been created as `XFS` previously but it was created as `EXT4` 
The expectation is that prior wrongly created volumes continue to function.

**Environment:**
 - Longhorn version: v1.2

**Additional context**
The fsType support was added in https://github.com/longhorn/longhorn-share-manager/pull/16
as part of the support for encrypted RWX volumes.

**Comment by joshimoo** [comment]
@innobead I plan on backporting this for v1.2.1 since it means previously incorrectly marked prior created RWX volumes no longer function. This can happen for manually created volumes as well as RWX volumes created as part of a storage class where the user specified `XFS` as the filesystem.

This only requires minor changes on the share-manager.

**Comment by yasker** [comment]
This sounds serious. How about previously created xfs RWO volumes?

**Comment by joshimoo** [comment]
@yasker this only impacts RWX volumes that have been marked as XFS, since there was no mechanism to transmit the users desired FS choice, so we always formatted as `EXT4`. 

In V1.2 we transmit the chosen FS based on the PV volumeParamters (comes from storage class or manually specified as part of PV creation) to the share-manager so it tries to mount the FS on the device based on the users chosen FS.

But since previously created volumes were always created as `EXT4` even if the user desired XFS as part of the storage class this will lead to a missmatch and therefore the mount will fail.

**Comment by joshimoo** [comment]
Testing steps:
- install v1.1.X
- create 2 rwx volumes + pvc via (storageclass-xfs, longhorn-ui) 
- create rwx workloads using the above volumes
- wait for success then verify fs is EXT4 (mount | grep /export) on the share-manager pod

- update to v1.2.X using this share-manager image `joshimoo/longhorn-share-manager:v_20210914`
- scale down -> up workloads
- verify workloads continue to function
- verify share-manager logs: `disk is already formatted to %v but user requested fs is %v using existing device fs type for mount`

**Comment by innobead** [comment]
ref: https://github.com/longhorn/longhorn-share-manager/pull/17#issuecomment-919001518

**Comment by khushboo-rancher** [comment]
Verified with Longhorn-master and v1.2.x - `09/16/2021`

Validation - **Passed**

I was able to reproducible the problem with v1.2.0. which is solved with v1.2.x and master.
Verified the newly created rwx volume with `xfs` PV, it is getting mounted as `xfs` in the share-manager

@joshimoo Just wanted to clarify the previously created `xfs` PV which got mounted as `ext4` will remain as `ext4`, we can't do anything for that?
